microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/commons/JavaCursorContext.java

```diff
@@ -0,0 +1,31 @@
+/*******************************************************************************
+* Copyright (c) 2023 Red Hat Inc. and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+* which is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lsp4mp.commons;
+
+/**
+ * Represents the context where the cursor is in a Java file.
+ */
+public enum JavaCursorContext {
+
+	EMPTY_FILE(1),
+	CLASS(2),
+	NONE_OF_THE_ABOVE(3);
+
+	int id;
+
+	private JavaCursorContext(int id) {
+		this.id = id;
+	}
+
```

Comment:
Define a `forValue` static method that LSP4J will use to serialize this enum like https://github.com/eclipse/lsp4mp/blob/caf91c68e7f2fe27072f1f947e3c5476f3289d8f/microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/commons/ClasspathKind.java#L42

Reply:
I don't seem to be having any problems serializing the enum, is this needed?

Reply:
Indeed it serialize the enum, but with name. If you provide `forValue` static method, LSP4J serializes the enum by using the int value to have a minimal JSON response size. 

You can see that any LSP4J enum provides this method. See for instance https://github.com/eclipse/lsp4j/blob/5605ca82252833777dbef82ca13a403dbfb9d906/org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/DiagnosticSeverity.java#L46

Reply:
The enum is represented as an integer in the JSON-RPC message without adding `forValue`.

Reply:
I see what you mean now, the enum is serialized as a number, but not as the same number